### PR TITLE
Generate R2R images in temp folder

### PR DESCRIPTION
**Pull request by gkhanna79**
This will avoid locking issues when generating R2R images.

@schellap @ramarag PTAL.

### CODE REVIEW

File: build_projects/shared-build-targets-utils/Utils/Crossgen.cs [hunk 1]
```diff
@@ -198,6 +198,9 @@ public void CrossgenDirectory(string sharedFxPath, string pathToAssemblies)
                 { "COMPlus_PartialNGen", "0" }
             };
 
+            // Form the dynamic path that would not collide if another instance of this is running.
+            string basePath = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
```
Comment (ellismg) [inline build_projects/shared-build-targets-utils/Utils/Crossgen.cs]: We will leak this directory, which seems less than ideal.

Comment (schellap) [inline build_projects/shared-build-targets-utils/Utils/Crossgen.cs]: +1. It may be better to keep this repo local? In obj?

Comment (gkhanna79) [inline build_projects/shared-build-targets-utils/Utils/Crossgen.cs]: Currently, it gets generated in the intermediates folder. I have added cleanup of the folder once the crossgen is complete.

**Comment by gkhanna79** [comment]
@schellap PTAL

File: build_projects/shared-build-targets-utils/Utils/Crossgen.cs (hunk 1 above)
Comment (ellismg) [inline build_projects/shared-build-targets-utils/Utils/Crossgen.cs]: +1 to @schellap's comment.  It should be simple enough as just not using Path.GetTempPath() and using obj folder, right?  You can still use the guid if there were races?

**Comment by gkhanna79** [comment]
I had seen the issue in intermediate folder. But for now, I am adding the Guid folder to it, so lets see if it still results in the locking problem.

PTAL.

**Comment by schellap** [comment]
Looks ideal to me!

**Pull request merged**